### PR TITLE
Adds var/needs_all_cures

### PR DESCRIPTION
**Pull request by LanCartwright**
Adds var var/needs_all_cures
Default =  TRUE.

What this does:
Cure = X, Y, Z
needs_all_cures = TRUE
All three chemicals are required to cure virus.

Cure = A, B, C
var/needs_all_cures = FALSE
One of the three chemicals are required to cure the virus.

### CODE REVIEW

File: code/datums/diseases/_disease.dm [hunk 1]
```diff
@@ -92,7 +92,7 @@ var/list/diseases = subtypesof(/datum/disease)
 
 	. = 1
 	for(var/C_id in cures)
-		if(!affected_mob.reagents.has_reagent(C_id))
+		if(!affected_mob.reagents.has_reagent(C_id) && needs_all_cures)
 			.--
 			break //One missing cure is enough to fail
```
Comment (duncathan) [inline code/datums/diseases/_disease.dm]: Replace the above with this:
```
. = cures.len
for(var/C_id in cures)
    if(!affected_mob.has_reagent(C_id))
        .--
if(!. || (needs_all_cures && . < cures.len))
    return 0
```
With tabs, not spaces

File: code/datums/diseases/_disease.dm [hunk 2]
```diff
@@ -90,12 +91,12 @@ var/list/diseases = subtypesof(/datum/disease)
...
-	. = 1
+	. = cures.len
 	for(var/C_id in cures)
-		if(!affected_mob.reagents.has_reagent(C_id))
+		if(!affected_mob.has_reagent(C_id))
 			.--
-			break //One missing cure is enough to fail
-
+	if(!. || (needs_all_cures && . < cures.len))
+	return 0
```
Comment (duncathan) [inline code/datums/diseases/_disease.dm]: Indent this one further

File: code/datums/diseases/_disease.dm [hunk 3]
```diff
@@ -61,8 +61,9 @@ var/list/diseases = subtypesof(/datum/disease)
 	var/permeability_mod = 1
 	var/severity =	NONTHREAT
 	var/list/required_organs = list()
-
+	var/needs_all_cures = TRUE
 	var/list/strain_data = list() //dna_spread special bullshit
+	var/cureCheck = 1
```
Comment (duncathan) [inline code/datums/diseases/_disease.dm]: ?

File: code/datums/diseases/_disease.dm (hunk 2 above)
Comment (MatthewWork) [inline code/datums/diseases/_disease.dm]: Bad indentation. Also, use neersighted's way instead. WAY easier to read.
https://www.irccloud.com/pastebin/K2ncYgIr/

**Pull request merged**